### DISCUSSION TITLE
What is difference between Application and Consumer?

### DISCUSSION THREAD

**Discussion by poo1882** [Kong/kong#9658] [Help] [answered]
Hi everyone, I'm new to Kong and I've tried to play around with Kong Manager and currently trying out OAuth plugin.
I found that the portal allows the developers to create an Application, after that they can use the generated client_id & client_secret to generate access token, and use those token to access Services, just like Consumer.

However, I'm not sure what is the difference between Application and Consumer? As in my use case I would need to assign scope to consumer, but my understanding is we cannot do that in Application level. So, I want my developers to register their application via Dev Portal and is there anyway to transform those Application into Consumer? or what am I miss understanding on something? 

please kindly suggest. Thank you so much.

**Comment by felderi** [comment, marked as answer]
Hi @poo1882!  Applications provide Developers the ability to, in a self-service way, get access to APIs managed by Kong with no interaction from the Kong team to generate credentials required.  

With Consumers, the Kong team creates consumers, generates credentials and needs to share them with the Developers that need access to the APIs.

You can think as Applications as a type of Consumer in Kong that allows Developers to automatically obtain credentials for and subscribe to the required APIs.

Let us know if you need additional clarification.  As a Kong customer, happy to discuss over a web session as well.  

Fel.

**Reply by teraflops-droid** [reply]
Hi @felderi There is a section for developers that can create api credentials right away with no application creation required. What is this api credentials for? Thank you in advance.

**Comment by poo1882** [comment]
Hi @felderi , thank you for the clarification.

However, I've one more question. Is there any way we can assign an Application to a Consumer Group?
